src/ui/components/common/vote_buttons.rs

```diff
@@ -41,6 +29,7 @@ pub fn VoteButtons<VA>(
   id: PostOrCommentId,
   score: Signal<i64>,
   vote_action: ServerAction<VA>,
+  #[prop(optional)] class: &'static str,
```

Comment:
I looked at [some of the docs for this](https://tailwind-fuse.vercel.app/), and it seems like these tailwind fuse classes are meant to be used to define various *styles*, not store extra data. 

So like `enum VoteStyle { None, FilledIn, Outlined }` or whatever it is.

This one also is confusing me because its an enum, but its being used like a component?

Reply:
> ...and it seems like these tailwind fuse classes are meant to be used to define various styles, not store extra data.

That's what I'm using it for, although it can be a bit difficult to grasp at first. I also struggled with it when I first used the library. I think, in the context of the vote buttons, it becomes clearer when you look at how it's used in this component.

Looking at the upvote button part of the `VoteButtons` component, we see this:

```rust
<button
    type="submit"
     class=move || {
         with!(
             | is_upvote | if * is_upvote { Vote::Up } else { Vote::None }.as_class().to_owned()
        )
    }

    title=move_tr!("upvote")
    disabled=move || !user_is_logged_in.get() || vote_action.pending().get()
>

    <Icon icon=IconType::Upvote />
</button>
```

Notice that `as_class` is called on the result of the conditional for `class` (which is a `VoteEnum` value). If this was expressed with using a string directly, it would look something like the following:

```rust
class=move || {
    with!(
        |is_upvote| tw_join!("align-bottom disabled:cursor-not-allowed disabled:text-neutral-content", if *is_upvote {"text-success"} else {"text-neutral"})
    )
}
```

I suppose those styles could be extracted to consts though, so it could end up looking something like this:

```rust
class=move || {
    with!(
        |is_upvote| tw_join!(vote_base_class, if *is_upvote {vote_success_class} else {no_vote_class})
    )
}
```

However, looking at it now, the `tw_join!` looks cleaner in this scenario, and the tailwind fuse enums/types seem to be best when they're being used as arguments. Alternatively, I could also try extracting the individual buttons into components, which could simplify the `VoteButtons` component and make the enums more useful. I'll play around with it a bit and see what looks better.

Reply:
I just refactored the vote buttons. Take a look and see if you like it.

Reply:
Looks much better.